scipy/conftest.py

```diff
@@ -403,9 +402,10 @@ def skip_or_xfail_xp_backends(request: pytest.FixtureRequest,
         request, skip_or_xfail=skip_or_xfail
     )
     xp = request.param
-    skip_or_xfail = getattr(pytest, skip_or_xfail)
-    reason = skip_xfail_reasons.get(xp.__name__)
-    if reason:
+    if xp.__name__ in skip_xfail_reasons:
```

Comment:
This was causing the test not to be skipped as we ended up with something like `{torch: None}` due to the explicit reason=None from make_skip_xp_backends.